cpp/include/cudf/detail/iterator.cuh

```diff
@@ -358,21 +361,7 @@ struct scalar_value_accessor {
                  "the data type mismatch");
   }
 
-  /**
-   * @brief returns the value of the scalar.
-   *
-   * @throw `cudf::logic_error` if this function is called in host.
-   *
-   * @return value of the scalar.
-   */
-  __device__ inline const Element operator()(size_type) const
-  {
-#if defined(__CUDA_ARCH__)
-    return dscalar.value();
-#else
-    CUDF_FAIL("unsupported device scalar iterator operation");
-#endif
-  }
+  __device__ inline Element const operator()(size_type) const { return dscalar.value(); }
```

Comment:
Does Doxygen complain about the removed documentation? Just making sure this isn't conflicting with @karthikeyann's efforts to fix Doxygen warnings.

Reply:
Related: https://github.com/rapidsai/cudf/pull/11003#discussion_r885819659

Reply:
I have no idea about doxygen warning. Still waiting for @karthikeyann to give comment (https://github.com/rapidsai/cudf/pull/11003#discussion_r885819659).

Reply:
Thanks @davidwendt. It sounds like it doesn't matter and I'm fine with removing these docstrings, so I'll resolve this.